pyteal/ast/itxn.py

```diff
@@ -200,6 +205,108 @@ def SetFields(cls, fields: Dict[TxnField, Union[Expr, List[Expr]]]) -> Expr:
         fieldsToSet = [cls.SetField(field, value) for field, value in fields.items()]
         return Seq(fieldsToSet)
 
+    @classmethod
+    def MethodCall(
+        cls,
+        *,
+        app_id: Expr,
+        method_signature: str,
+        args: list[abi.BaseType | Expr | dict[TxnField, Expr | list[Expr]]],
+        fields: dict[TxnField, Expr | list[Expr]],
+    ) -> Expr:
```

Comment:
We'll definitely want a comprehensive docstring explaining how to use this function

Reply:
Started something here but I expect there should be more and possible improve formatting for docs